master/static/migrations/20230908100300_fix-dupe-checkpoints.tx.up.sql

```diff
@@ -0,0 +1,54 @@
+DROP VIEW proto_checkpoints_view;
+DROP VIEW checkpoints_view;
+
+CREATE OR REPLACE VIEW checkpoints_view AS
+    SELECT
+        c.id AS id,
+        c.uuid AS uuid,
+        c.task_id,
+        c.allocation_id,
+        c.report_time,
+        c.state,
+        c.resources,
+        c.metadata,
+        t.id AS trial_id,
+        e.id AS experiment_id,
+        e.config AS experiment_config,
+        t.hparams AS hparams,
+        s.metrics AS training_metrics,
+        v.metrics->'validation_metrics' AS validation_metrics,
+        (v.metrics->'validation_metrics'->>(e.config->'searcher'->>'metric'))::float8 AS searcher_metric,
+        CAST(c.metadata->>'steps_completed' AS int) as steps_completed,
+        c.size
+    FROM checkpoints_v2 AS c
+    LEFT JOIN trial_id_task_id AS task ON c.task_id = task.task_id
+    LEFT JOIN trials AS t on t.id = task.trial_id
+    LEFT JOIN experiments AS e on t.experiment_id = e.id
+    LEFT JOIN raw_validations AS v on CAST(c.metadata->>'steps_completed' AS int) = v.total_batches and t.id = v.trial_id AND NOT v.archived
+    LEFT JOIN raw_steps AS s on CAST(c.metadata->>'steps_completed' AS int) = s.total_batches and t.id = s.trial_id AND NOT s.archived;
```

Comment:
these are the changes, but reviewer should still check that this doesn't revert something